File: java/src/main/java/io/amerhwitat/bizxtreme/GameLauncher.java
package io.amerhwitat.bizxtreme;

import io.amerhwitat.bizxtreme.api.BizXtremeApi;
import io.amerhwitat.bizxtreme.game.TycoonGame;

/** Single Java entry point for starting BizXtreme or its Tycoon mode. */
public final class GameLauncher {
    private GameLauncher() {}
    public static void main(String[] args) {
        BizXtremeApi api = new BizXtremeApi();
        System.out.println("BizXtreme game starting");
        System.out.println(api.health());
        if (args.length > 0 && "tycoon".equalsIgnoreCase(args[0])) {
            TycoonGame game = new TycoonGame(10_000);
            System.out.println("BizXtreme Tycoon ready: cash=" + game.cash());
        }
    }
}

File: python/bizxtreme/game_launcher.py
"""Single Python entry point for starting BizXtreme and Tycoon mode."""

from .api import BizXtremeApi
from .game.tycoon import TycoonGame


def main(mode: str = "default") -> int:
    api = BizXtremeApi()
    print("BizXtreme game starting")
    print(api.health())
    if mode.lower() == "tycoon":
        game = TycoonGame(cash=10_000)
        print(f"BizXtreme Tycoon ready: cash={game.cash}")
    return 0


if __name__ == "__main__":
    import sys
    raise SystemExit(main(sys.argv[1] if len(sys.argv) > 1 else "default"))
